### ISSUE TITLE
MAINT: inconsistent use of rng in classifier objects

### ISSUE THREAD

**Issue by sdtemple** [lanl/GFDL#126]
Is there a reason why the first instance of `self.W_append` in lines 581-585 and 642 - 646 of `model.py` creates a separate random number generator different from `rng`? I would think you want all random generation to be controlled by the same `rng`.

This seems inconsistent with the `sklearn` pull https://github.com/scikit-learn/scikit-learn/pull/33215

```python
self.W_.append(
    self._weight_mode(
        self._N, hidden_layer_sizes[0], rng=self.get_generator(self.seed)
        )
    )
self.b_.append(
    self._weight_mode(1, hidden_layer_sizes[0], rng=rng)
    .reshape(-1)
    )

for i, layer in enumerate(hidden_layer_sizes[1:]):
    # (n_hidden, n_features)
    self.W_.append(
        self._weight_mode(hidden_layer_sizes[i] + self._N, layer, rng=rng)
        )
    # (n_hidden,)
    self.b_.append(
        self._weight_mode(1, layer, rng=rng,).reshape(-1)
        )
```

The suggested fix is 
```diff
+    self._weight_mode(
+       self._N, hidden_layer_sizes[0], rng=rng
+       )
-   self._weight_mode(
-       self._N, hidden_layer_sizes[0], rng=self.get_generator(self.seed)
-       )

**Comment by tylerjereddy** [comment]
Those lines are for `EnsembleGFDL`, which isn't part of the `sklearn` thrust, but I agree that it doesn't look optimal--there do seem to be more calls to `get_generator()`, which in turn calls `np.random.default_rng(seed)`, than there should be. Could probably be cleaned up a bit at least. Possibly the class instance should create a single `self.rng` somewhere, depending on how `sklearn` typically does it for their estimators.

It may be possible to demonstrate that this is a bug, since even with a fixed seed the advancing bit stream has consequences:

```python
>>> import numpy as np
>>> rng = np.random.default_rng(5)
>>> rng.random(3)
array([0.80500292, 0.80794079, 0.51532556])
>>> rng.random(3)
array([0.28580138, 0.0539307 , 0.38336888])
>>> rng.random(3)
array([0.40847321, 0.04527519, 0.04875771])
>>> rng.random(3)
array([0.99917612, 0.65236911, 0.2345102 ])
```

So a fix for this may warrant some regression testing as well. For example, testing that interwoven calls to `fit()` and `partial_fit()` on the same class instance draw from the same seed rather than at least partially drawing from a new bit generator as currently happens, I believe. May be a little annoying to test, but should be doable since it affects the random matrix stuff at least.

**Comment by sdtemple** [comment]
I could look into this after the IC proposal deadline.

**Comment by eiviani-lanl** [comment]
This was a little hack we had to do to match GrafoRVFLs results. If we would like to pull grafo out of the test suite, then I agree it would be optimal to perform seeding as is done in the sklearn PR.
